src/DonationForms/Blocks/DonationFormBlock/resources/editor/components/EntitySelector.tsx

```diff
@@ -0,0 +1,87 @@
+// components/EntitySelector.tsx
+
+import { useState } from 'react';
+import { __ } from '@wordpress/i18n';
+import ReactSelect from 'react-select';
+import { reactSelectStyles, reactSelectThemeStyles } from '../styles/reactSelectStyles';
+import logo from '../images/givewp-logo.svg';
+
+import '../styles/index.scss';
+import usePostState from "../hooks/usePostState";
+import {useEffect} from "@wordpress/element";
+
+export type EntityOption = {
+    label: string;
+    value: number | string;
+};
+
+type EntitySelectorProps = {
+    id: string;
+    label: string;
+    options: EntityOption[];
+    isLoading: boolean;
+    emptyMessage: string;
+    loadingMessage: string;
+    onConfirm: (id: number | string) => void;
+    buttonText?: string;
+    disabled?: boolean;
+};
+
+// @ts-ignore
+const savePost = () => dispatch('core/editor').savePost();
+
+export default function EntitySelector({
```

Comment:
This is a nifty component! Since this is more of a multi-purpose component I would elevate it to be reusable outside this block's domian.